### ISSUE TITLE
Attempt to index field 'evoGUI' (a nil value)

### ISSUE THREAD

**Issue by AesSedai** [narc0tiq/evoGUI#71]
Factorio version 0.15.22
EvoGui version 0.5.0

Chat log reads:
[evoGUI|on_tick:update_gui] Error: __Evo|GUI__/evoGUI.lua:101: attempt to index a field 'evoGUI' (a nil value)

This was in an existing game, where EvoGUI was present from the start. Issue occurred when Factorio updated from 0.15.21 to 0.15.22. Changing the Factorio version back to 0.15.21 stops the error from occurring.

**Comment by narc0tiq** [comment]
I'm mobile at the moment, and it's late here, so the fix will have to wait until tomorrow. But thank you for the report!

**Comment by Draconomial** [comment]
👍

**Comment by KaiserTom** [comment]
Important to note that upon starting a new game in 0.15.22 it results in no issues with evoGUI, it works as intended. However anytime you load the save, it results in the error seen above.

**Comment by narc0tiq** [comment]
So it turns out Factorio is destroying EvoGUI's GUI on each load, for whatever reason, so I just changed the update method to recreate it if necessary (YARM already had this).

Expect a release momentarily.

**Comment by narc0tiq** [comment]
And done and published; use the in-game updater and you should get a nice working EvoGUI.

**Comment by narc0tiq** [comment]
Hey, folks, if you're still watching this bug, I found that it's actually [a bug in Factorio 0.15.22](https://forums.factorio.com/viewtopic.php?f=7&t=50055) which Rseding91 has fixed and kindly shared the details of. Given that, I was able to make a workaround, but I need one or two people other than myself to test and make sure I didn't break anything anew.

So, I need literally anyone who already has EvoGUI installed to try out [the new beta release, v0.4.205](https://github.com/narc0tiq/evoGUI/releases/tag/v0.4.205), and report back here if it's working or if anything broke. Once I get one or two more positive reports, I can release it to the mod portal, as well.
